### PR TITLE
Rounded corners follow smart_gaps

### PR DESCRIPTION
**Pull request by jhpaques**
Implementation of #53.

### CODE REVIEW

**Review by ErikReider — APPROVED** [review]
LGTM!

**Comment by ErikReider** [comment]
According to the man 5 page, `inverse_outer` should only enable outer gaps when one window is visible but it seems to be broken even in upstream sway

File: sway/tree/workspace.c [hunk 1]
```diff
@@ -861,6 +861,10 @@ void workspace_add_gaps(struct sway_workspace *ws) {
 		ws->current_gaps.right = 0;
 		ws->current_gaps.bottom = 0;
 		ws->current_gaps.left = 0;
+        for (int i = 0; i < ws->tiling->length; ++i) {
+            struct sway_container *con = ws->tiling->items[i];
+            con->corner_radius = 0;
+        }
```
Comment (ErikReider) [inline sway/tree/workspace.c:867]: Follow indent style: Tabs

File: sway/tree/workspace.c [hunk 2]
```diff
@@ -870,8 +874,20 @@ void workspace_add_gaps(struct sway_workspace *ws) {
...
```
Comment (ErikReider) [inline sway/tree/workspace.c:880]: Follow indent style: Tabs

File: sway/tree/workspace.c [hunk 3]
```diff
@@ -870,8 +874,20 @@ void workspace_add_gaps(struct sway_workspace *ws) {
...
 	} else {
 		ws->current_gaps = ws->gaps_outer;
+        for (int i = 0; i < ws->tiling->length; ++i) {
+            struct sway_container *con = ws->tiling->items[i];
+            con->corner_radius = config->corner_radius;
+        }
+        for (int i = 0; i < ws->floating->length; ++i) {
+            struct sway_container *con = ws->floating->items[i];
+            con->corner_radius = config->corner_radius;
+        }
```
Comment (ErikReider) [inline sway/tree/workspace.c:890]: Follow indent style: Tabs

Comment (ErikReider) [inline sway/tree/workspace.c:890]: Is this needed? If so, why?

**Review by ErikReider — CHANGES_REQUESTED** [review]
Sorry, missed some stuff

File: sway/tree/workspace.c (hunk 3 above)
Comment (jhpaques) [reply sway/tree/workspace.c:890]: When a window is opened in floating mode and is the first on the workspace, it loses its rounded corners without this.

**Comment by ErikReider** [comment]
> > According to the man 5 page, `inverse_outer` should only enable outer gaps when one window is visible but it seems to be broken even in upstream sway
> 
> I think this can be easily fixed by changing the place of a bracket. Should I make a PR upstream?

@WillPower3309 what's your opinion on this?

**Comment by WillPower3309** [comment]
> > > According to the man 5 page, `inverse_outer` should only enable outer gaps when one window is visible but it seems to be broken even in upstream sway
> > 
> > I think this can be easily fixed by changing the place of a bracket. Should I make a PR upstream?
> 
> @WillPower3309 what's your opinion on this? I'd do that but Will's the boss man

Yep, I think that's a good idea! I'll take a quick look at this PR when I can after work

**Comment by jhpaques** [comment]
> > According to the man 5 page, `inverse_outer` should only enable outer gaps when one window is visible but it seems to be broken even in upstream sway
> 
> I think this can be easily fixed by changing the place of a bracket. Should I make a PR upstream?

Actually, I was mistaken, `inverse_outer` is not broken, but `gaps outer` should be set for it to work.

**Comment by WillPower3309** [comment]
This is really great stuff, but is making me realize that I may have to rethink how we define a container's corner_radius, as it shouldn't need to be reset back to the config default every time we add the gaps. I'm going to think about how to address this and get back to you on Friday

**Comment by WillPower3309** [comment]
Thanks a ton for the PR @jhpaques ! I really appreciate the time you spent on this, however, I recently changed things around with render.c to make this feature much more easy to implement, and more efficient as well. As such, I've implemented this feature in #62 , and will be closing this PR.

**Pull request closed without merging**